File: 4_Kyu_Kata/Sum_Strings_as_Numbers/Python/sum_strings_as_numbers.test.py
import codewars_test as test
from sum_strings_as_numbers import sumStrings


@test.describe("Test Cases")
def test_group():
    @test.it("Fixed Tests")
    def test_case():
        test.assert_equals(sumStrings("8797", "45"), "8842")
        test.assert_equals(sumStrings("800", "9567"), "10367")
        test.assert_equals(sumStrings("99", "1"), "100")
        test.assert_equals(sumStrings("00103", "08567"), "8670")
        test.assert_equals(sumStrings("", "5"), "5")
        test.assert_equals(
            sumStrings("712569312664357328695151392", "8100824045303269669937"), "712577413488402631964821329"
        )
        test.assert_equals(
            sumStrings("50095301248058391139327916261", "81055900096023504197206408605"), "131151201344081895336534324866"
        )

File: kata-4-kyu/sum-strings-as-numbers/javascript/sumStringsAsNumbers.Test.js
//+ ====================================================================================================================
//+
//+ 4 kyu - Sum Strings as Numbers  [ ID: 5324945e2ece5e1f32000370 ] (sum-strings-as-numbers)
//+ URL: https://www.codewars.com/kata/5324945e2ece5e1f32000370
//+ Category: ALGORITHMS  |  Tags: STRINGS | BIG INTEGERS | ALGORITHMS
//+
//+ ====================================================================================================================

import { assert } from "vitest"
const { sumStrings } = require("./sumStringsAsNumbers")

describe("Tests", () => {
  it("test", () => {
    function msg(a, b) {
      if (typeof returned !== "string") {
        return "Function must return a string value"
      } else {
        return "sumStrings('" + a + "', '" + b + "')"
      }
    }
    function t(a, b, ans) {
      const returned = sumStrings(a, b)
      assert.strictEqual(returned, ans, msg(a, b, returned))
    }
    t("123", "456", "579")

    t("8797", "45", "8842")
    t("800", "9567", "10367")
    t("99", "1", "100")
    t("00103", "08567", "8670")
    t("", "5", "5")
    t("712569312664357328695151392", "8100824045303269669937", "712577413488402631964821329")
    t("50095301248058391139327916261", "81055900096023504197206408605", "131151201344081895336534324866")

    const BigNumber = require("bignumber.js")

    function sumStringsSol(a, b) {
      if (!a) a = "0"
      if (!b) b = "0"
      return new BigNumber(a)
        .plus(new BigNumber(b))
        .toFixed(0)
    }

    for (let i = 0; i < 50; i++) {
      let a = BigNumber.random(30).toString().slice(2),
        b = BigNumber.random(30).toString().slice(2),
        ans = sumStringsSol(a, b)
      t(a, b, ans)
    }
  })
})
